src/npc.cpp

```diff
@@ -1455,7 +1455,15 @@ void npc::decide_needs()
         elem = 20;
     }
     if( weapon.is_gun() ) {
-        needrank[need_ammo] = 5 * get_ammo( ammotype( *weapon.type->gun->ammo.begin() ) ).size();
+        int ups_drain = weapon.get_gun_ups_drain();
+        if( ups_drain > 0 ) {
+            int ups_charges = charges_of( "UPS_off", ups_drain ) +
+                              charges_of( "UPS_off", ups_drain );
+            needrank[need_ammo] = ups_charges / ups_drain;
```

Comment:
Need to do something with clang tidy here.

```
/home/travis/build/CleverRaven/Cataclysm-DDA/src/npc.cpp:1462:35: error: result of integer division used in a floating point context; possible loss of precision [bugprone-integer-division,-warnings-as-errors]

            needrank[need_ammo] = ups_charges / ups_drain;
```

Reply:
silently upconvert you silly machine!  I'll add the explicit cast to shut it up.